Add support for skipped tests

In robotframework 4.0.0 they added the `skip` keyword which results in a skipped result. But xunit2rst doesn't handle this correctly.

The skip reason is missing from the rest output and the result is failed.

The input XML file is:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="Suite" tests="3" errors="0" failures="1" skipped="1" time="0.044">
<testcase classname="Suite" name="passed test" time="0.002">
</testcase>
<testcase classname="Suite" name="failed test" time="0.002">
<failure message="'False' should be true." type="AssertionError"/>
</testcase>
<testcase classname="Suite" name="skipped test" time="0.001">
<skipped message="This test doesn't work on this setup" type="SkipExecution"/>
</testcase>
</testsuite>
```

The output rst file looks like:
```rst
.. _integration_test_report_example_itest:

=======================================================
Integration Test Report for example_itest
=======================================================

The log file that contains details about the executed test cases can be found `here <../example_log.html>`_.

.. contents:: `Contents`
    :depth: 2
    :local:


Test Cases
==========

.. item:: REPORT_ITEST_EXAMPLE-PASSED_TEST Test report for ITEST_EXAMPLE-PASSED_TEST
    :passes: ITEST_EXAMPLE-PASSED_TEST

    Test result: Pass

.. item:: REPORT_ITEST_EXAMPLE-FAILED_TEST Test report for ITEST_EXAMPLE-FAILED_TEST
    :fails: ITEST_EXAMPLE-FAILED_TEST

    Test result: Fail

    ::

      AssertionError: 'False' should be true.

.. item:: REPORT_ITEST_EXAMPLE-SKIPPED_TEST Test report for ITEST_EXAMPLE-SKIPPED_TEST
    :fails: ITEST_EXAMPLE-SKIPPED_TEST

    Test result: Fail

Traceability Matrix
===================

The below table traces the test report to test cases.

.. item-matrix:: Linking these integration test reports to integration test cases
    :source: REPORT_ITEST_EXAMPLE-
    :target: ITEST_EXAMPLE-
    :sourcetitle: Integration test report
    :targettitle: Integration test specification
    :type: fails passes
    :stats:
    :group: top
    :nocaptions:
```

While I was expecting something like the following for the skipped test.

```rst
.. item:: REPORT_ITEST_EXAMPLE-SKIPPED_TEST Test report for ITEST_EXAMPLE-SKIPPED_TEST
    :skipped: ITEST_EXAMPLE-SKIPPED_TEST

    Test result: Skipped

    ::

        This test doesn't work on this setup
```

--- a/README.rst
+++ b/README.rst
@@ -108,7 +108,7 @@ Test Result of Itemized Suites
 ------------------------------
 
 Pass (passes)
-  No test cases have failed.
+  No test cases have failed. Some might be skipped.
 
 Fail (fails)
   At least one test case has failed.

--- a/README.rst
+++ b/README.rst
@@ -135,8 +135,8 @@ Its value should start with *u* or *i* and gets parsed case-insensitively.
 Include Message of Failure(s)
 =============================
 
-When the ``-f, --failure-message`` flag is set, the tool includes the messages of all failures of the item's test cases
-in its body.
+When the ``-f, --failure-message`` flag is set, the tool includes the messages of all failures and reasons for skipping
+of the item's test cases in its body.
 
 Links to Log File
 =================

--- a/README.rst
+++ b/README.rst
@@ -102,7 +102,19 @@ Itemization
 
 By default, all *testcase* elements from the input file are used to created treaceability items. This may not always be
 desired. The ``-s, --itemize-suites`` flag lets the script itemize *testsuite* elements instead. In this case, the
-*testcase* elements will still be parsed to determine whether the testsuite passed or failed.
+*testcase* elements will still be parsed to determine whether the testsuite passed or failed or skipped.
+
+Test Result of Itemized Suites
+------------------------------
+
+Pass (passes)
+  No test cases have failed.
+
+Fail (fails)
+  At least one test case has failed.
+
+Skip (skipped)
+  All test cases have been skipped.
 
 Item IDs
 ========

--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -84,6 +84,7 @@ pygments_style = 'sphinx'
 traceability_relationships = {
     'passes': 'passed_by',
     'fails': 'failed_by',
+    'skipped': 'skipped_by',
     'validates': 'validated_by',
     'ext_robotframeworklog': '',
 }
@@ -92,6 +93,8 @@ traceability_relationship_to_string = {
     'passed_by': 'Passed by',
     'fails': 'Fails',
     'failed_by': 'Failed by',
+    'skipped': 'Skipped',
+    'skipped_by': 'Skipped by',
     'validates': 'Validates',
     'validated_by': 'Validated by',
     'ext_robotframeworklog': 'Details in log file',

--- a/doc/source/robot/example.robot
+++ b/doc/source/robot/example.robot
@@ -22,6 +22,11 @@ Another test
     ${sum}             Evaluate  ${128} + ${128}
     Should Be Equal    ${sum}    ${256}
 
+A skipped test
+    [Documentation]    Example of a test that is always skipped
+    [Tags]             RQT-LOGGING
+    Skip               This test case is always skipped.
+
 *** Keywords ***
 My Keyword
     [Arguments]    ${path}

--- a/mlx/xunit2rst.mako
+++ b/mlx/xunit2rst.mako
@@ -77,21 +77,25 @@ ${generate_item(test_name, relationship, failure_message, [test], (len(suite_nam
         % endfor
     % else:  # create traceable item per testsuite element
 <%
-test_result = 'Skip'
-relationship = 'skipped'
+test_result = 'Pass'
+relationship = 'passes'
 test_name = _convert_name(suite.attrib['name'])
 # skip testsuite elements that have no testcase element (typically the first testsuite element only)
 if not len(suite):
     continue
 
+skipped_counter = 0
 for test in suite:
     if test.findall('failure'):
         test_result = 'Fail'
         relationship = 'fails'
         break
-    if not len(test):
-        test_result = 'Pass'
-        relationship = 'passes'
+    if test.findall('skipped'):
+        skipped_counter += 1
+else:
+    if skipped_counter == len(suite):
+        test_result = 'Skip'
+        relationship = 'skipped'
 %>\
 ${generate_item(test_name, relationship, failure_message, suite, (0, suite_idx))}\
     % endif

--- a/mlx/xunit2rst.mako
+++ b/mlx/xunit2rst.mako
@@ -66,7 +66,6 @@ else:
     relationship = 'passes'
 if add_links:
     class_name = test.attrib.get('classname', '')
-    print(class_name)
     if class_name.startswith(f"{suite.attrib.get('name')}."):
         suite_name = class_name.split('.')[-1]
         if suite_name not in suite_names:

--- a/mlx/xunit2rst.mako
+++ b/mlx/xunit2rst.mako
@@ -55,8 +55,12 @@ test_idx = 0
 <%
 test_name = _convert_name(test.attrib['name'])
 if len(test):
-    test_result = 'Fail'
-    relationship = 'fails'
+    if test.findall('skipped'):
+        test_result = 'Skipped'
+        relationship = 'skipped'
+    else:
+        test_result = 'Fail'
+        relationship = 'fails'
 else:
     test_result = 'Pass'
     relationship = 'passes'
@@ -82,10 +86,13 @@ if not len(suite):
     continue
 
 for test in suite:
-    if len(test):
+    if test.findall('failure'):
         test_result = 'Fail'
         relationship = 'fails'
         break
+    if test.findall('skipped'):
+        test_result = 'Skipped'
+        relationship = 'skipped'
 %>\
 ${generate_item(test_name, relationship, failure_message, suite, (0, suite_idx))}\
     % endif
@@ -116,7 +123,7 @@ The below table traces the test report to test cases.
 <% prepend_literal_block = True %>
 % if failure_msg:
     % for test in tests:
-        % for failure in test.iterfind('failure'):
+        % for failure in test.findall('failure') + test.findall('skipped'):
             % if prepend_literal_block:
     ::
 <% prepend_literal_block = False %>

--- a/mlx/xunit2rst.mako
+++ b/mlx/xunit2rst.mako
@@ -77,25 +77,21 @@ ${generate_item(test_name, relationship, failure_message, [test], (len(suite_nam
         % endfor
     % else:  # create traceable item per testsuite element
 <%
-test_result = 'Pass'
-relationship = 'passes'
+test_result = 'Skip'
+relationship = 'skipped'
 test_name = _convert_name(suite.attrib['name'])
 # skip testsuite elements that have no testcase element (typically the first testsuite element only)
 if not len(suite):
     continue
 
-skipped_counter = 0
 for test in suite:
     if test.findall('failure'):
         test_result = 'Fail'
         relationship = 'fails'
         break
-    if test.findall('skipped'):
-        skipped_counter += 1
-else:
-    if skipped_counter == len(suite):
-        test_result = 'Skip'
-        relationship = 'skipped'
+    if not len(test):
+        test_result = 'Pass'
+        relationship = 'passes'
 %>\
 ${generate_item(test_name, relationship, failure_message, suite, (0, suite_idx))}\
     % endif

--- a/mlx/xunit2rst.mako
+++ b/mlx/xunit2rst.mako
@@ -84,12 +84,16 @@ test_name = _convert_name(suite.attrib['name'])
 if not len(suite):
     continue
 
+skipped_counter = 0
 for test in suite:
     if test.findall('failure'):
         test_result = 'Fail'
         relationship = 'fails'
         break
     if test.findall('skipped'):
+        skipped_counter += 1
+else:
+    if skipped_counter == len(suite):
         test_result = 'Skip'
         relationship = 'skipped'
 %>\
@@ -120,7 +124,7 @@ The below table traces the test report to test cases.
 
     Test result: ${test_result}
 <% prepend_literal_block = True %>
-% if failure_msg:
+% if failure_msg and relationship != 'passes':
     % for test in tests:
         % for failure in test.findall('failure') + test.findall('skipped'):
             % if prepend_literal_block:

--- a/mlx/xunit2rst.mako
+++ b/mlx/xunit2rst.mako
@@ -56,7 +56,7 @@ test_idx = 0
 test_name = _convert_name(test.attrib['name'])
 if len(test):
     if test.findall('skipped'):
-        test_result = 'Skipped'
+        test_result = 'Skip'
         relationship = 'skipped'
     else:
         test_result = 'Fail'
@@ -90,7 +90,7 @@ for test in suite:
         relationship = 'fails'
         break
     if test.findall('skipped'):
-        test_result = 'Skipped'
+        test_result = 'Skip'
         relationship = 'skipped'
 %>\
 ${generate_item(test_name, relationship, failure_message, suite, (0, suite_idx))}\

--- a/setup.py
+++ b/setup.py
@@ -7,7 +7,7 @@ setup(
     use_scm_version={
         'write_to': 'mlx/__xunit2rst_version__.py'
     },
-    setup_requires=['setuptools-scm'],
+    setup_requires=['setuptools-scm>=6.0.0,<7.*'],
     url='https://github.com/melexis/xunit2rst',
     license='Apache License Version 2.0',
     author='JasperCraeghs',

--- a/tests/acceptance_test.py
+++ b/tests/acceptance_test.py
@@ -278,13 +278,26 @@ def test_log_file_links_multiple_suites():
 
 @with_setup(setup)
 def test_support_for_skipped_junit():
-    '''Tests based on utest reports in JUnit format - test support for skipped tests '''
+    '''Tests based on utest reports in JUnit format - test support for skipped tests'''
     file_name = 'utest_my_lib_report_skipped'
     rst_file_name = '{}.rst'.format(file_name)
     xml_file_name = '{}.xml'.format(file_name)
     input_xml = str(TEST_IN_DIR / xml_file_name)
     output_rst = str(TEST_OUT_DIR / rst_file_name)
     xunit2rst_check(input_xml, output_rst, itemize_suites=False, prefix='')
+
+    reference_rst = str(TEST_IN_DIR / rst_file_name)
+    assert filecmp.cmp(output_rst, reference_rst)
+
+
+@with_setup(setup)
+def test_support_for_skipped_itemize_suites():
+    '''Tests based on utest reports in JUnit format - test support for skipped tests when itemizing test suites'''
+    rst_file_name = '{}.rst'.format('utest_my_lib_suites_report_skipped')
+    xml_file_name = '{}.xml'.format('utest_my_lib_report_skipped')
+    input_xml = str(TEST_IN_DIR / xml_file_name)
+    output_rst = str(TEST_OUT_DIR / rst_file_name)
+    xunit2rst_check(input_xml, output_rst, itemize_suites=True, prefix='', failures=True)
 
     reference_rst = str(TEST_IN_DIR / rst_file_name)
     assert filecmp.cmp(output_rst, reference_rst)

--- a/tests/acceptance_test.py
+++ b/tests/acceptance_test.py
@@ -276,5 +276,18 @@ def test_log_file_links_multiple_suites():
     assert filecmp.cmp(output_rst, reference_rst)
 
 
+@with_setup(setup)
+def test_support_for_skipped():
+    '''Tests based on itest reports in xUnit format - including failure messages and a test that was skipped'''
+    rst_file_name = '{}.rst'.format('itest_report_skipped_failures_log')
+    xml_file_name = '{}.xml'.format('itest_report_skipped')
+    input_xml = str(TEST_IN_DIR / xml_file_name)
+    output_rst = str(TEST_OUT_DIR / rst_file_name)
+    xunit2rst_check(input_xml, output_rst, failures=True)
+
+    reference_rst = str(TEST_IN_DIR / rst_file_name)
+    assert filecmp.cmp(output_rst, reference_rst)
+
+
 if __name__ == '__main__':
     nose.main()

--- a/tests/acceptance_test.py
+++ b/tests/acceptance_test.py
@@ -277,7 +277,21 @@ def test_log_file_links_multiple_suites():
 
 
 @with_setup(setup)
-def test_support_for_skipped():
+def test_support_for_skipped_junit():
+    '''Tests based on utest reports in JUnit format - test support for skipped tests '''
+    file_name = 'utest_my_lib_report_skipped'
+    rst_file_name = '{}.rst'.format(file_name)
+    xml_file_name = '{}.xml'.format(file_name)
+    input_xml = str(TEST_IN_DIR / xml_file_name)
+    output_rst = str(TEST_OUT_DIR / rst_file_name)
+    xunit2rst_check(input_xml, output_rst, itemize_suites=False, prefix='')
+
+    reference_rst = str(TEST_IN_DIR / rst_file_name)
+    assert filecmp.cmp(output_rst, reference_rst)
+
+
+@with_setup(setup)
+def test_support_for_skipped_xunit():
     '''Tests based on itest reports in xUnit format - including failure messages and a test that was skipped'''
     rst_file_name = '{}.rst'.format('itest_report_skipped_failures_log')
     xml_file_name = '{}.xml'.format('itest_report_skipped')

--- a/tests/test_in/itest_report_skipped.xml
+++ b/tests/test_in/itest_report_skipped.xml
@@ -1,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="Example" tests="3" errors="0" failures="1" skipped="0" time="0.049">
+<testcase classname="Example" name="First Test" time="0.002">
+<failure message="Directory 'C:\nonexistent' does not exist." type="AssertionError"></failure>
+</testcase>
+<testcase classname="Example" name="An Unlinked Test" time="0.002">
+</testcase>
+<testcase classname="Example" name="Another test" time="0.002">
+<skipped message="This test doesn't work on this setup" type="SkipExecution"/>
+</testcase>
+<properties>
+    <property name="Documentation" value="My test suite title" />
+    <property name="More Info" value="Test support for xUnit output of RobotFramework==5.0.0" />
+    <property name="Version" value="1.0" />
+</properties>
+</testsuite>

--- a/tests/test_in/itest_report_skipped_failures_log.rst
+++ b/tests/test_in/itest_report_skipped_failures_log.rst
@@ -1,0 +1,52 @@
+.. _integration_test_report_itest_report_skipped_failures_log:
+
+=============================================================
+Integration Test Report for itest_report_skipped_failures_log
+=============================================================
+
+
+.. contents:: `Contents`
+    :depth: 2
+    :local:
+
+
+Test Cases
+==========
+
+.. item:: REPORT_ITEST-FIRST_TEST Test report for ITEST-FIRST_TEST
+    :skipped: ITEST-FIRST_TEST
+
+    Test result: Skipped
+
+    ::
+
+      AssertionError: Directory 'C:\nonexistent' does not exist.
+
+.. item:: REPORT_ITEST-AN_UNLINKED_TEST Test report for ITEST-AN_UNLINKED_TEST
+    :passes: ITEST-AN_UNLINKED_TEST
+
+    Test result: Pass
+
+.. item:: REPORT_ITEST-ANOTHER_TEST Test report for ITEST-ANOTHER_TEST
+    :skipped: ITEST-ANOTHER_TEST
+
+    Test result: Skipped
+
+    ::
+
+      SkipExecution: This test doesn't work on this setup
+
+Traceability Matrix
+===================
+
+The below table traces the test report to test cases.
+
+.. item-matrix:: Linking these integration test reports to integration test cases
+    :source: REPORT_ITEST-
+    :target: ITEST-
+    :sourcetitle: Integration test report
+    :targettitle: Integration test specification
+    :type: fails passes
+    :stats:
+    :group: top
+    :nocaptions:

--- a/tests/test_in/itest_report_skipped_failures_log.rst
+++ b/tests/test_in/itest_report_skipped_failures_log.rst
@@ -30,7 +30,7 @@ Test Cases
 .. item:: REPORT_ITEST-ANOTHER_TEST Test report for ITEST-ANOTHER_TEST
     :skipped: ITEST-ANOTHER_TEST
 
-    Test result: Skipped
+    Test result: Skip
 
     ::
 

--- a/tests/test_in/itest_report_skipped_failures_log.rst
+++ b/tests/test_in/itest_report_skipped_failures_log.rst
@@ -14,9 +14,9 @@ Test Cases
 ==========
 
 .. item:: REPORT_ITEST-FIRST_TEST Test report for ITEST-FIRST_TEST
-    :skipped: ITEST-FIRST_TEST
+    :fails: ITEST-FIRST_TEST
 
-    Test result: Skipped
+    Test result: Fail
 
     ::
 

--- a/tests/test_in/utest_my_lib_report_skipped.rst
+++ b/tests/test_in/utest_my_lib_report_skipped.rst
@@ -14,9 +14,9 @@ Test Cases
 ==========
 
 .. item:: REPORT_UTEST_MY_LIB-MY_FUNCTION_SUCCESS Test report for UTEST_MY_LIB-MY_FUNCTION_SUCCESS
-    :passes: UTEST_MY_LIB-MY_FUNCTION_SUCCESS
+    :skipped: UTEST_MY_LIB-MY_FUNCTION_SUCCESS
 
-    Test result: Pass
+    Test result: Skip
 
 .. item:: REPORT_UTEST_MY_LIB-TEST_MY_FUNCTION_NOT_ERASED Test report for UTEST_MY_LIB-TEST_MY_FUNCTION_NOT_ERASED
     :skipped: UTEST_MY_LIB-TEST_MY_FUNCTION_NOT_ERASED
@@ -28,10 +28,15 @@ Test Cases
 
     Test result: Pass
 
-.. item:: REPORT_UTEST_MY_LIB-SOME_FUNCTION Test report for UTEST_MY_LIB-SOME_FUNCTION
-    :fails: UTEST_MY_LIB-SOME_FUNCTION
+.. item:: REPORT_UTEST_MY_LIB-SOME_FUNCTION1 Test report for UTEST_MY_LIB-SOME_FUNCTION1
+    :fails: UTEST_MY_LIB-SOME_FUNCTION1
 
     Test result: Fail
+
+.. item:: REPORT_UTEST_MY_LIB-SOME_FUNCTION2 Test report for UTEST_MY_LIB-SOME_FUNCTION2
+    :skipped: UTEST_MY_LIB-SOME_FUNCTION2
+
+    Test result: Skip
 
 Traceability Matrix
 ===================

--- a/tests/test_in/utest_my_lib_report_skipped.rst
+++ b/tests/test_in/utest_my_lib_report_skipped.rst
@@ -1,0 +1,49 @@
+.. _unit_test_report_utest_my_lib_report_skipped:
+
+================================================
+Unit Test Report for utest_my_lib_report_skipped
+================================================
+
+
+.. contents:: `Contents`
+    :depth: 2
+    :local:
+
+
+Test Cases
+==========
+
+.. item:: REPORT_UTEST_MY_LIB-MY_FUNCTION_SUCCESS Test report for UTEST_MY_LIB-MY_FUNCTION_SUCCESS
+    :passes: UTEST_MY_LIB-MY_FUNCTION_SUCCESS
+
+    Test result: Pass
+
+.. item:: REPORT_UTEST_MY_LIB-TEST_MY_FUNCTION_NOT_ERASED Test report for UTEST_MY_LIB-TEST_MY_FUNCTION_NOT_ERASED
+    :skipped: UTEST_MY_LIB-TEST_MY_FUNCTION_NOT_ERASED
+
+    Test result: Skip
+
+.. item:: REPORT_UTEST_MY_LIB-TEST_MY_FUNCTION_NOT_UNLOCKED Test report for UTEST_MY_LIB-TEST_MY_FUNCTION_NOT_UNLOCKED
+    :passes: UTEST_MY_LIB-TEST_MY_FUNCTION_NOT_UNLOCKED
+
+    Test result: Pass
+
+.. item:: REPORT_UTEST_MY_LIB-SOME_FUNCTION Test report for UTEST_MY_LIB-SOME_FUNCTION
+    :fails: UTEST_MY_LIB-SOME_FUNCTION
+
+    Test result: Fail
+
+Traceability Matrix
+===================
+
+The below table traces the test report to test cases.
+
+.. item-matrix:: Linking these unit test reports to unit test cases
+    :source: REPORT_UTEST_MY_LIB-
+    :target: UTEST_MY_LIB-
+    :sourcetitle: Unit test report
+    :targettitle: Unit test specification
+    :type: fails passes
+    :stats:
+    :group: top
+    :nocaptions:

--- a/tests/test_in/utest_my_lib_report_skipped.xml
+++ b/tests/test_in/utest_my_lib_report_skipped.xml
@@ -1,18 +1,23 @@
 <?xml version='1.0' encoding='utf-8'?>
 <testsuites>
     <testsuite errors="0" failures="0" name="MY_LIB" skipped="0" tests="0" timestamp="2019-11-29T09:58:22" />
-    <testsuite errors="0" failures="0" name="MY_LIB.UTEST_MY_LIB-MY_FUNCTION_SUCCESS" skipped="0" tests="1" time="0.0" timestamp="2019-11-29T09:58:22">
-        <testcase classname="MY_LIB.UTEST_MY_LIB-MY_FUNCTION_SUCCESS" name="UTEST_MY_LIB-MY_FUNCTION_SUCCESS" time="0.0" />
+    <testsuite errors="0" failures="0" name="MY_LIB.UTEST_MY_LIB-MY_FUNCTION_SUCCESS" skipped="1" tests="1" time="0.0" timestamp="2019-11-29T09:58:22">
+        <testcase classname="MY_LIB.UTEST_MY_LIB-MY_FUNCTION_SUCCESS" name="UTEST_MY_LIB-MY_FUNCTION_SUCCESS" time="0.0">
+            <skipped message="Skipping the sole test case in this test suite." />
+        </testcase>
     </testsuite>
     <testsuite errors="0" failures="0" name="MY_LIB.UTEST_MY_LIB-MY_FUNCTION_LOCKED" skipped="1" tests="2" time="0.0" timestamp="2019-11-29T09:58:22">
         <testcase classname="MY_LIB.UTEST_MY_LIB-MY_FUNCTION_LOCKED" name="UTEST_MY_LIB-test_my_function_not_erased" time="0.0">
-            <skipped message="Testing support for skipped tests." />
+            <skipped message="Skipping this test case but the other test case in this suite should pass." />
         </testcase>
         <testcase classname="MY_LIB.UTEST_MY_LIB-MY_FUNCTION_LOCKED" name="UTEST_MY_LIB-test_my_function_not_unlocked" time="0.0" />
     </testsuite>
-    <testsuite errors="0" failures="1" name="MY_LIB.UTEST_MY_LIB-SOME_FUNCTION" skipped="0" tests="1" time="0.0" timestamp="2019-11-29T09:59:30">
-        <testcase classname="MY_LIB.UTEST_MY_LIB-SOME_FUNCTION" name="UTEST_MY_LIB-SOME_FUNCTION" time="0.0">
+    <testsuite errors="0" failures="1" name="MY_LIB.UTEST_MY_LIB-SOME_FUNCTION" skipped="1" tests="2" time="0.0" timestamp="2019-11-29T09:59:30">
+        <testcase classname="MY_LIB.UTEST_MY_LIB-SOME_FUNCTION" name="UTEST_MY_LIB-SOME_FUNCTION1" time="0.0">
             <failure message="File: ./unit_test/my_functions.c&#10;Line: 49&#10;Message: exp &quot;12 34 56 &quot; was &quot;12 34 99 &quot;&#10;" />
+        </testcase>
+        <testcase classname="MY_LIB.UTEST_MY_LIB-SOME_FUNCTION" name="UTEST_MY_LIB-SOME_FUNCTION2" time="0.0">
+            <skipped message="Skipping this test case but the other test case in this suite should fail." />
         </testcase>
     </testsuite>
 </testsuites>

--- a/tests/test_in/utest_my_lib_report_skipped.xml
+++ b/tests/test_in/utest_my_lib_report_skipped.xml
@@ -1,0 +1,18 @@
+<?xml version='1.0' encoding='utf-8'?>
+<testsuites>
+    <testsuite errors="0" failures="0" name="MY_LIB" skipped="0" tests="0" timestamp="2019-11-29T09:58:22" />
+    <testsuite errors="0" failures="0" name="MY_LIB.UTEST_MY_LIB-MY_FUNCTION_SUCCESS" skipped="0" tests="1" time="0.0" timestamp="2019-11-29T09:58:22">
+        <testcase classname="MY_LIB.UTEST_MY_LIB-MY_FUNCTION_SUCCESS" name="UTEST_MY_LIB-MY_FUNCTION_SUCCESS" time="0.0" />
+    </testsuite>
+    <testsuite errors="0" failures="0" name="MY_LIB.UTEST_MY_LIB-MY_FUNCTION_LOCKED" skipped="1" tests="2" time="0.0" timestamp="2019-11-29T09:58:22">
+        <testcase classname="MY_LIB.UTEST_MY_LIB-MY_FUNCTION_LOCKED" name="UTEST_MY_LIB-test_my_function_not_erased" time="0.0">
+            <skipped message="Testing support for skipped tests." />
+        </testcase>
+        <testcase classname="MY_LIB.UTEST_MY_LIB-MY_FUNCTION_LOCKED" name="UTEST_MY_LIB-test_my_function_not_unlocked" time="0.0" />
+    </testsuite>
+    <testsuite errors="0" failures="1" name="MY_LIB.UTEST_MY_LIB-SOME_FUNCTION" skipped="0" tests="1" time="0.0" timestamp="2019-11-29T09:59:30">
+        <testcase classname="MY_LIB.UTEST_MY_LIB-SOME_FUNCTION" name="UTEST_MY_LIB-SOME_FUNCTION" time="0.0">
+            <failure message="File: ./unit_test/my_functions.c&#10;Line: 49&#10;Message: exp &quot;12 34 56 &quot; was &quot;12 34 99 &quot;&#10;" />
+        </testcase>
+    </testsuite>
+</testsuites>

--- a/tests/test_in/utest_my_lib_suites_report_skipped.rst
+++ b/tests/test_in/utest_my_lib_suites_report_skipped.rst
@@ -23,7 +23,7 @@ Test Cases
       Skipping the sole test case in this test suite.
 
 .. item:: REPORT_UTEST_MY_LIB-MY_FUNCTION_LOCKED Test report for UTEST_MY_LIB-MY_FUNCTION_LOCKED
-    :skipped: UTEST_MY_LIB-MY_FUNCTION_LOCKED
+    :passes: UTEST_MY_LIB-MY_FUNCTION_LOCKED
 
     Test result: Pass
 

--- a/tests/test_in/utest_my_lib_suites_report_skipped.rst
+++ b/tests/test_in/utest_my_lib_suites_report_skipped.rst
@@ -1,0 +1,58 @@
+.. _unit_test_report_utest_my_lib_suites_report_skipped:
+
+=======================================================
+Unit Test Report for utest_my_lib_suites_report_skipped
+=======================================================
+
+
+.. contents:: `Contents`
+    :depth: 2
+    :local:
+
+
+Test Cases
+==========
+
+.. item:: REPORT_UTEST_MY_LIB-MY_FUNCTION_SUCCESS Test report for UTEST_MY_LIB-MY_FUNCTION_SUCCESS
+    :skipped: UTEST_MY_LIB-MY_FUNCTION_SUCCESS
+
+    Test result: Skip
+
+    ::
+
+      Skipping the sole test case in this test suite.
+
+.. item:: REPORT_UTEST_MY_LIB-MY_FUNCTION_LOCKED Test report for UTEST_MY_LIB-MY_FUNCTION_LOCKED
+    :skipped: UTEST_MY_LIB-MY_FUNCTION_LOCKED
+
+    Test result: Skip
+
+    ::
+
+      Skipping this test case but the other test case in this suite should pass.
+
+.. item:: REPORT_UTEST_MY_LIB-SOME_FUNCTION Test report for UTEST_MY_LIB-SOME_FUNCTION
+    :fails: UTEST_MY_LIB-SOME_FUNCTION
+
+    Test result: Fail
+
+    ::
+
+      File: ./unit_test/my_functions.c Line: 49 Message: exp "12 34 56 " was "12 34 99 "
+
+      Skipping this test case but the other test case in this suite should fail.
+
+Traceability Matrix
+===================
+
+The below table traces the test report to test cases.
+
+.. item-matrix:: Linking these unit test reports to unit test cases
+    :source: REPORT_UTEST_MY_LIB-
+    :target: UTEST_MY_LIB-
+    :sourcetitle: Unit test report
+    :targettitle: Unit test specification
+    :type: fails passes
+    :stats:
+    :group: top
+    :nocaptions:

--- a/tests/test_in/utest_my_lib_suites_report_skipped.rst
+++ b/tests/test_in/utest_my_lib_suites_report_skipped.rst
@@ -25,11 +25,7 @@ Test Cases
 .. item:: REPORT_UTEST_MY_LIB-MY_FUNCTION_LOCKED Test report for UTEST_MY_LIB-MY_FUNCTION_LOCKED
     :skipped: UTEST_MY_LIB-MY_FUNCTION_LOCKED
 
-    Test result: Skip
-
-    ::
-
-      Skipping this test case but the other test case in this suite should pass.
+    Test result: Pass
 
 .. item:: REPORT_UTEST_MY_LIB-SOME_FUNCTION Test report for UTEST_MY_LIB-SOME_FUNCTION
     :fails: UTEST_MY_LIB-SOME_FUNCTION

--- a/tox.ini
+++ b/tox.ini
@@ -51,6 +51,7 @@ deps=
     mlx.traceability >= 4.3.2
     mlx.warnings >= 1.2.0
     mlx.robot2rst >= 2.0.2
+    robotframework >= 5.0.1
 whitelist_externals =
     bash
     make

